app/code/community/Ebizmarts/MailChimp/Model/Api/Subscribers/MailchimpTags.php

```diff
@@ -588,11 +588,13 @@ protected function addCompany($customAtt, $customer, $key)
     protected function getLastOrderByEmail()
     {
         $lastOrder = $this->getLastOrder();
+
         if ($lastOrder === null) {
             $helper = $this->getMailchimpHelper();
             $orderCollection = $helper->getOrderCollectionByCustomerEmail($this->getSubscriber()->getSubscriberEmail());
+
             if ($this->isNotEmptyOrderCollection($orderCollection)) {
-                $lastOrder = $orderCollection->setOrder('created_at', 'DESC')->getFirstItem();
+                $lastOrder = $orderCollection->setOrder('created_at', 'DESC')->setPageSize(1, 1)->getLastItem();
```

Comment:
Move setPageSize filter to the collection definition to keep everything together.
setPageSize function only takes 1 parameter.

Reply:
Solved.